.github/workflows/test.yml

```diff
@@ -36,6 +36,35 @@ jobs:
         pytest tests --doctest-modules serde -v
         cd examples && python runner.py
 
+  check_formatting:
+    name: Check formatting
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+    - name: Restore pip cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.pythonLocation }}
+        key: ubuntu-latest-${{ env.pythonLocation }}-${{ hashFiles('**/.pre-commit-config.yaml', '**/pyproject.toml') }}
+    - name: Install dependencies
+      run: |
+        pip install pre-commit
+        pre-commit install
+    - name: Check formatting
+      run: pre-commit run -a
+    - name: Comment PR
+      if: ${{ failure() && github.event_name == 'pull_request' }}
+      uses: thollander/actions-comment-pull-request@master
+      with:
+        message: 'Please consider formatting your code according to the standards described here: <link to the CONTRIBUTION.md>'
```

Comment:
I feel we need CONTRIBUTING.md .. 🤔

Reply:
Sure we need it, I just wanted to show the concept to you and get your approval :-) I will add CONTRIBUTING.md tomorrow and update this PR.